### PR TITLE
fix(job-ctrl): refactor pod and service

### PR DESCRIPTION
**Pull request by diaozhongpu**
refactor pod and service under controller, by extracting method

### CODE REVIEW

File: controllers/elasticdl/pod.go [hunk 1]
```diff
@@ -66,23 +65,13 @@ func (r *ElasticDLJobReconciler) GetPodsForJob(obj interface{}) ([]*corev1.Pod,
...
 		return fmt.Errorf("%+v is not a type of ElasticDLJob", job)
 	}
 
-	log.Info("Deleting pod", "controller name", r.ControllerName(), "pod name", pod.Namespace+"/"+pod.Name)
-	if err := r.Delete(context.Background(), pod); err != nil && !errors.IsNotFound(err) {
-		r.recorder.Eventf(elasticdlJob, corev1.EventTypeWarning, job_controller.FailedDeletePodReason, "Error deleting: %v", err)
-		return err
-	}
-	r.recorder.Eventf(elasticdlJob, corev1.EventTypeNormal, job_controller.SuccessfulDeletePodReason, "Deleted pod: %v", pod.Name)
+	r.ctrl.BroadcastDeletePod(elasticdlJob, pod)
```
Comment (jian-he) [inline controllers/elasticdl/pod.go:75]: why name Broadcast ?

File: pkg/tensorboard/tensorboard.go [hunk 2]
```diff
@@ -220,7 +220,7 @@ func syncPod(jc job_controller.JobController, c client.Client, metaObj metav1.Ob
 		}
 	}
 
-	if err := jc.CreatePod(metaObj, tbRT, index, template, false); err != nil {
+	if err := jc.CreateCommonPod(metaObj, tbRT, index, template, false); err != nil {
```
Comment (jian-he) [inline pkg/tensorboard/tensorboard.go:223]: CreateCommonPod -> createPodReplica

File: pkg/test_job/v1/test_job_controller.go [hunk 3]
```diff
@@ -104,7 +104,7 @@ func (t *TestJobController) DeleteService(job interface{}, name string, namespac
 	return nil
 }
 
-func (t *TestJobController) CreatePod(job interface{}, pod *corev1.Pod) error {
+func (in *TestJobController) CreatePod(job interface{}, pod *corev1.Pod) error {
```
Comment (jian-he) [inline pkg/test_job/v1/test_job_controller.go:107]: unnecessary change, remove

File: pkg/job_controller/test_job_controller.go [hunk 4]
```diff
@@ -74,6 +74,10 @@ func (t *TestJobController) GetJobFromAPIClient(namespace, name string) (v1.Obje
 	return t.job, nil
 }
 
+func (t TestJobController) CreatePod(job interface{}, pod *corev1.Pod) error {
```
Comment (jian-he) [inline pkg/job_controller/test_job_controller.go:77]: unnecessary change ? remove

File: controllers/elasticdl/pod.go (hunk 1 above)
Comment (jian-he) [reply controllers/elasticdl/pod.go:75]: 就叫deletePod吧, similar to others

File: pkg/job_controller/pod.go [hunk 5]
```diff
@@ -318,6 +318,11 @@ func (jc *JobController) ReconcilePods(
 	return nil
 }
 
+// CreatePod creates the pod
+func (jc *JobController) CreatePod(job interface{}, pod *v1.Pod) error {
```
Comment (SimonCqk) [inline pkg/job_controller/pod.go:322]: is it necessary to abstract a new function since argument `job` is not referenced ?

File: pkg/job_controller/api/v1/interface.go [hunk 6]
```diff
@@ -50,13 +50,13 @@ type ControllerInterface interface {
 	UpdateJobStatusInApiServer(job interface{}, jobStatus *JobStatus) error
 
 	// CreateService creates the service
-	CreateService(job interface{}, service *corev1.Service) error
+	// CreateService(job interface{}, service *corev1.Service) error
 
 	// DeleteService deletes the service
 	DeleteService(job interface{}, name string, namespace string) error
```
Comment (SimonCqk) [inline pkg/job_controller/api/v1/interface.go:56]: `DeleteService` and `DeletePod` can be safely removed as well

File: pkg/job_controller/job.go [hunk 7]
```diff
@@ -47,11 +47,15 @@ func (jc *JobController) deletePodsAndServices(runPolicy *apiv1.RunPolicy, job i
 		if *runPolicy.CleanPodPolicy == apiv1.CleanPodPolicyRunning && pod.Status.Phase != v1.PodRunning {
 			continue
 		}
-		if err := jc.Controller.DeletePod(job, pod); err != nil {
+		job, ok := job.(runtime.Object)
```
Comment (SimonCqk) [inline pkg/job_controller/job.go:50]: rename left value `job` to `runtimeObj` to avoid variable confusion
